gson/src/main/java/com/google/gson/JsonParser.java

```diff
@@ -27,6 +27,45 @@
 /**
  * A parser to parse JSON into a parse tree of {@link JsonElement}s.
  *
+ * <p>The JSON data is parsed in {@linkplain JsonReader#setStrictness(Strictness) lenient mode}.
+ *
+ * <p>Here's an example of parsing from a string:
+ *
+ * <pre>
+ * String json = "{\"key\": \"value\"}";
+ * JsonElement jsonElement = JsonParser.parseString(json);
+ * JsonObject jsonObject = jsonElement.getAsJsonObject();
+ * </pre>
+ *
+ * <p>It can also parse from a reader:
+ *
+ * <pre>
+ * try (Reader reader = new FileReader("my-data.json", StandardCharsets.UTF_8)) {
+ *   JsonElement jsonElement = JsonParser.parseReader(reader);
+ *   JsonObject jsonObject = jsonElement.getAsJsonObject();
+ * }
+ * </pre>
+ *
+ * <p>If you want to parse from a {@link JsonReader} for more customized parsing requirements, the
+ * following example demonstrates how to achieve it:
+ *
+ * <pre>
+ * String json = "{\"skipObj\": {\"skipKey\": \"skipValue\"},\"obj\": {\"key\": \"value\"}}";
```

Comment:
Really minor, but maybe for pretty-printing consistency and to make it a bit easier to read, there should be a space after the `,`:
```suggestion
 * String json = "{\"skipObj\": {\"skipKey\": \"skipValue\"}, \"obj\": {\"key\": \"value\"}}";
```